neps/nep-0452.md

```diff
@@ -0,0 +1,239 @@
+/// * The assets *MUST* be sent to the `new_account_id` passed in.
+/// * The newly created account *MUST* have a new access key added to its account (either
+///   full or limited access) in the same receipt that the account was created in.
+/// * The Public key must be in a binary format with base58 string serialization with human-readable curve.
+///   The key types currently supported are secp256k1 and ed25519. Ed25519 public keys accepted are 32 bytes
+///   and secp256k1 keys are the uncompressed 64 format.
+///
+/// Arguments:
+/// * `new_account_id`: the valid NEAR account which is being created and should
+///   receive the linkdrop assets
+/// * `new_public_key`: the valid public key that should be used for the access key added to the newly created account (serialized with borsh).
+///
+/// Returns `true` if the claim was successful meaning the `new_account_id` was created and all assets were sent to it.
+function create_account_and_claim(new_account_id: string, new_public_key: string) -> Promise<boolean>;
+```
+
+## Reference Implementation
+
+Below are some references for linkdrop contracts.
+
+- [Link Drop Contract](https://github.com/near/near-linkdrop)
+
+- [Keypom Contract](https://github.com/keypom/keypom)
+
+## Security Implications
+
+1. Linkdrop Creation
+   Linkdrop creation involves creating keypairs that, when used, have access to assets such as $NEAR, FTs, NFTs, etc. These keys should be limited access and restricted to specific functionality. For example, they should only have permission to call `claim` and `create_account_and_claim`. Since the keys allow the holder to sign transactions on behalf of the linkdrop contract, without the proper security measures, they could be used in a malicious manner (for example executing private methods or owner-only functions).
+
+Another important security implication of linkdrop creation is to ensure that only one key is mapped to a set of assets at any given time. Externally, assets such as FTs, and NFTs belong to the overall linkdrop contract account rather than a specific access key. It is important to ensure that specific keys can only claim assets that they are mapped to.
+
+2. Linkdrop Key Management  
+   Key management is a critical safety component of linkdrops. The linkdrop contract should implement a key management strategy for keys such that a reentrancy attack does not occur. For example, one strategy may be to "lock" or mark a key as "in transfer" such that it cannot be used again until the transfer is complete.
+
+3. Asset Refunds & Failed Claims
+   Given that linkdrops could contain multiple different assets such as NFTs, or fungible tokens, sending assets might happen across multiple blocks. If the claim was unsuccessful (such as passing in an invalid account ID), it is important to ensure that all state is properly managed and assets are optionally refunded depending on the linkdrop contract's implementation.
+
+4. Fungible Tokens & Future Data
```

Comment:
```suggestion
4. Storage Registration & Future Data
```

Reply:
This is not specific to Fungible Tokens. Can also be related to NFTs or Multitoken... Let's just call it Storage Registration (which is a standard by it's own).